dask/array/tests/test_array_utils.py

```diff
@@ -48,6 +48,11 @@ def test_meta_from_array(asarray):
     assert meta_from_array("foo") == "foo"
     assert meta_from_array(np.dtype("float32")) == np.dtype("float32")
 
+    assert meta_from_array("").dtype == np.array("").dtype
+    assert meta_from_array("", dtype="bool").dtype == np.array([], dtype="bool").dtype
+    assert meta_from_array("", dtype="int").dtype == np.array([], dtype="int").dtype
+    assert meta_from_array("", dtype="float").dtype == np.array([], dtype="float").dtype
```

Comment:
Could you also test `"str"`, `u""`, and `u"str"` here

Reply:
Are the filled string cases (e.g., `"str"` and `u"str"`) actually happening in your tests? I'm asking this because I would generally expect `meta` to be reduced to an empty array (empty literal in this case), if that's not actually happening we may have a different issue.

Reply:
Nevermind, the test you're suggesting is from _before_ `meta` getting reduced, so it makes sense.